### PR TITLE
Add documentation describing package projects

### PR DESCRIPTION
**Pull request by ericstj**
/cc @weshaggard @chcosta

### CODE REVIEW

File: Documentation/coding-guidelines/package-projects.md [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+
+Asset projects (`.csproj`, `.vbproj`, or `.depproj`) can control their `{targetMoniker}` using the `PackageTargetFramework` property in the project file.  Similarly `{rid}` is controlled using the `PackageTargetRuntime` property.  In the corefx repo we automatically select default values for these properties based on the [Build pivots](#build-pivots).  These can be overridden in the project reference using metadata of the same name, but this is rarely needed.
+
+The primary thing that the library author needs to do in order to ensure the correct asset selection is:
+1. Configure the correct projects in your library's `.builds` file.
```
Comment (weshaggard) [inline Documentation/coding-guidelines/package-projects.md]: This list doesn't show up correctly in github markdown.

File: Documentation/coding-guidelines/package-projects.md [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+2. Reference the `.builds` file from the package project.
+3. Provide a default PackageTargetFramework for empty-TargetGroup builds.
+```
+<PackageTargetFramework Condition="'$(PackageTargetFramework)' == ''">dotnet5.4</PackageTargetFramework>
+```
+
+### Desktop facades
```
Comment (weshaggard) [inline Documentation/coding-guidelines/package-projects.md]: To help clarify can you call this .NET Framework Facades as opposed to desktop.

File: Documentation/coding-guidelines/package-projects.md [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+Asset projects (`.csproj`, `.vbproj`, or `.depproj`) can control their `{targetMoniker}` using the `PackageTargetFramework` property in the project file.  Similarly `{rid}` is controlled using the `PackageTargetRuntime` property.  In the corefx repo we automatically select default values for these properties based on the [Build pivots](#build-pivots).  These can be overridden in the project reference using metadata of the same name, but this is rarely needed.
+
+The primary thing that the library author needs to do in order to ensure the correct asset selection is:
+1. Configure the correct projects in your library's `.builds` file.
+2. Reference the `.builds` file from the package project.
+3. Provide a default PackageTargetFramework for empty-TargetGroup builds.
+```
+<PackageTargetFramework Condition="'$(PackageTargetFramework)' == ''">dotnet5.4</PackageTargetFramework>
```
Comment (weshaggard) [inline Documentation/coding-guidelines/package-projects.md]: Without reading all this and just looking at this property it is unclear if this goes in the pkgproj or the csproj.

File: Documentation/coding-guidelines/package-projects.md [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+```
```
Comment (weshaggard) [inline Documentation/coding-guidelines/package-projects.md]: We need to add some information about how to select which dotnet/netstandard version to pick.

File: Documentation/coding-guidelines/package-projects.md [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+
+### Desktop facades
+Desktop facades must be part of the reference package.  This is because if we were to use the reference assembly on desktop it would have type collisions with whatever types already exist in the desktop reference assemblies.  Since we include the desktop reference facade in the reference package we also include the runtime facade in the same package for compression savings.
+
+## Applicability validation
+Part of package build is to ensure that a package is applicable on all platforms it supports and not applicable on platforms it does not support.  We do this validation for a set of targets established in the packaging tools (see [DefaultValidateFramework](https://github.com/dotnet/buildtools/blob/9f4ddda1cb021c9bd25f606bc4e74b92e4b82869/src/Microsoft.DotNet.Build.Tasks.Packaging/src/PackageFiles/Packaging.targets#L709)).  Package projects identify the targets supported in one of two ways.
+1. **Preferred:** Through `SupportedFramework` metadata on the project reference.  The metadata will associate the API version of that project reference with the frameworks listed.
```
Comment (weshaggard) [inline Documentation/coding-guidelines/package-projects.md]: This list isn't formatting correctly on github either.

**Comment by weshaggard** [comment]
Thanks for adding this documentation. A couple other things that I think we should cover (not necessarily as part of this PR).

- Versioning rules for both the libraries and packages. When to bump the major, minor, patch, etc. 
- How to select the correct NetStandard/Generation

**Comment by ericstj** [comment]
@dotnet-bot test  Innerloop Windows_NT Release Build and Test

File: Documentation/coding-guidelines/package-projects.md (hunk 1 above)
Comment (ericstj) [inline Documentation/coding-guidelines/package-projects.md]: fixing.  Looks like I have to include an additional newline.

File: Documentation/coding-guidelines/package-projects.md (hunk 2 above)
Comment (ericstj) [inline Documentation/coding-guidelines/package-projects.md]: sure

File: Documentation/coding-guidelines/package-projects.md (hunk 3 above)
Comment (ericstj) [inline Documentation/coding-guidelines/package-projects.md]: I'll clarify.

File: Documentation/coding-guidelines/package-projects.md (hunk 4 above)
Comment (ericstj) [inline Documentation/coding-guidelines/package-projects.md]: Added a section on this.

File: Documentation/coding-guidelines/package-projects.md (hunk 5 above)
Comment (ericstj) [inline Documentation/coding-guidelines/package-projects.md]: I needed to indent the source blocks.

**Comment by weshaggard** [comment]
What about the versioning strategy?

**Comment by ericstj** [comment]
That should be a separate doc.  I'm working on it now.

**Pull request merged**